marlowe-playground-client/src/MonadApp.purs

```diff
@@ -228,27 +231,40 @@ updateContractInStateP :: String -> MarloweState -> MarloweState
 updateContractInStateP text state = case runParser' (parseTerm contract) text of
   Right pcon ->
     let
-      (Tuple duplicates holes) = validateHoles $ getHoles mempty pcon
+      lintResult = lint pcon
+
+      lintHoles = view L._holes lintResult
+
+      (Holes holes) = lintHoles
+
+      warnings =
+        map (warningToAnnotation text "The contract can make a negative payment here") (view L._negativePayments lintResult)
+          <> map (warningToAnnotation text "The contract can make a negative deposit here") (view L._negativeDeposits lintResult)
 
       mContract = fromTerm pcon
     in
       case mContract of
         Just contract -> do
-          set _editorErrors [] <<< set _contract (Just contract) $ state
+          set _editorErrors warnings <<< set _contract (Just contract) $ state
         Nothing -> do
           let
-            holes' = fromFoldable $ Map.values holes
-
-            (Holes m) = getHoles mempty pcon
+            holes' = fold $ fromFoldable $ Map.values holes
 
-            holesm = getHoles mempty pcon
+            holesm = lintHoles
 
-            holes'' = fold $ fromFoldable $ Map.values m
+            holes'' = fold $ fromFoldable $ Map.values holes
```

Comment:
Hmm....can you skip the `fromFoldable` here? 🤔

Reply:
Also, `holes'` and `holes''` are identical aren't they? I think this section needs a tidying pass.